From b882013421036a05ea8c909292cc38d05fc29c89 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 5 Feb 2019 14:32:20 -0800
Subject: [PATCH] chromium: Exclude CRC32 for 32bit arm

This fixes build issues during cross compiling for linux using clang

Fixes
src/3rdparty/chromium/third_party/zlib/crc32_simd.c:184: undefined reference to `__crc32d'

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 chromium/third_party/zlib/BUILD.gn | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/chromium/third_party/zlib/BUILD.gn b/chromium/third_party/zlib/BUILD.gn
index 05907b481ac..dbc4be143cd 100644
--- a/chromium/third_party/zlib/BUILD.gn
+++ b/chromium/third_party/zlib/BUILD.gn
@@ -70,8 +70,11 @@ config("zlib_arm_crc32_config") {
     #  - ChromeOS has wrapper scripts that are borking the compiler flags.
     #  - Fuchsia just added a syscall for feature detection.
     # TODO(cavalcantii): crbug.com/810125.
+    defines = []
     if (!is_ios && !is_chromeos && !is_fuchsia) {
-      defines = [ "CRC32_ARMV8_CRC32" ]
+      if (current_cpu == "arm64") {
+        defines += [ "CRC32_ARMV8_CRC32" ]
+      }
       if (is_android) {
         defines += [ "ARMV8_OS_ANDROID" ]
       } else if (is_linux || is_chromeos) {
